### PR TITLE
Key position

### PR DESCRIPTION
**Pull request by wetda3**
-Key now choses a random square on the grid.
-Key now scales with grid

### CODE REVIEW

File: Scripts/key.gd [hunk 1]
```diff
@@ -14,3 +15,15 @@ func _process(delta):
 	
 func _physics_process(delta):
 	pass
+
+
+func _on_our_tile_grid_proportions(tile_scale, grid_size_in_tiles, tile_size):
+	
+	#first choses random x,y in tiles then converts that to position
+	var off_set = tile_size/2
```
Comment (werydude) [inline Scripts/key.gd:23]: `offset` is one word

File: Scripts/key.gd [hunk 2]
```diff
@@ -14,3 +15,15 @@ func _process(delta):
...
+	var rand_pos_x = key_rng.randi_range(0,grid_size_in_tiles[0]) 
+	var rand_pos_y =  key_rng.randi_range(0,grid_size_in_tiles[1]) 
+	scale = tile_scale
+	position = Vector2(rand_pos_x,rand_pos_y)*Vector2(tile_size) + Vector2(off_set)
```
Comment (werydude) [inline Scripts/key.gd:27]: This conversation should all be done in a system in the future

File: Scripts/key.gd [hunk 3]
```diff
@@ -14,3 +15,15 @@ func _process(delta):
...
+	print(position)
```
Comment (werydude) [inline Scripts/key.gd:28]: All "dev helpers" should be `print_debug`. Also, we have a stack tracer...

File: Scripts/key.gd [hunk 4]
```diff
@@ -14,3 +15,15 @@ func _process(delta):
...
 func _physics_process(delta):
 	pass
+
+
+func _on_our_tile_grid_proportions(tile_scale, grid_size_in_tiles, tile_size):
+	
+	#first choses random x,y in tiles then converts that to position
+	var off_set = tile_size/2
+	var rand_pos_x = key_rng.randi_range(0,grid_size_in_tiles[0]) 
+	var rand_pos_y =  key_rng.randi_range(0,grid_size_in_tiles[1])
```
Comment (werydude) [inline Scripts/key.gd:25]: I think we should use `randfn` here. 
'True' randomness can create frustrating repeated sequences, whereas normal distributions gives you an expectation of the kind games you'll have, and how often.

The primary consequence is that the corners of the screen will not be seen as often, but I think that's expected and desired. If not, it also ***gives us control*** over these things. [](edited-with-Axolo)

**Review by werydude — DISMISSED** [review]
Mostly good, just minor changes + new random function.

File: Scripts/key.gd (hunk 4 above)
Comment (wetda3) [reply Scripts/key.gd:25]: Keeping it true random

**Review by werydude — APPROVED** [review]

**Pull request closed without merging**